File: task_equivalents/llm_translations/starcoder/base/r/data_manipulation_preprocessing/02_merge_aggregate_reshape.R
library(dplyr) # for data manipulation
library(readr) # for reading CSV files
library(caret) # for data splitting
College <- read_csv("data/csv/College.csv")
school_names <- College$X1
rownames(College) <- school_names
College <- select(College, -X1)
College$Private01 <- ifelse(College$Private == "Yes", 1, 0)
College <- select(College, -Private)
X <- select(College, -Apps)
Y <- College$Apps
set.seed(1) # Set a seed for reproducibility
train_index <- sample(1:nrow(X), 0.5 * nrow(X))
xtrain <- X[train_index, ]
ytrain <- Y[train_index]
xtest <- X[-train_index, ]
ytest <- Y[-train_index]

File: task_equivalents/llm_translations/starcoder/base/python/data_manipulation_preprocessing/02_merge_aggregate_reshape.py
import pandas as pd

College = pd.read_csv('data/csv/College.csv', header=0)
College.set_index(College.columns[0], inplace=True)
College['Private01'] = (College['Private'] == 'Yes').astype(int)
College.drop(columns='Private', inplace=True)
X = College.drop(columns='Apps').apply(pd.to_numeric)
Y = College['Apps']

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.5, random_state=1)